overlays/bsp/u-boot/imx8qm/patches/0001-Add-support-for-Display-VM.patch

```diff
@@ -0,0 +1,64 @@
+From 89b4c386a476a53fc4c3b83ae896affc54cd2a58 Mon Sep 17 00:00:00 2001
+From: Juan Pablo Ruiz <juanpablo.ruiz@tii.ae>
+Date: Tue, 20 Dec 2022 11:47:22 +0000
+Subject: [PATCH 1/1] Add support for Display VM
+
+U-boot passes the imx8qm-mek-kvm-host.dtb to the host Linux. This DTB
+has the GPU/HDMI disabled on host and reserved for the VMs.
+
+U-boot loads the SCU configuration before HDMI firmware loading,
+by modifying the CONFIG_BOOTCOMMAND
+
+Disable video on uboot by disabling CONFIG_DM_VIDEO. The U-boot uses
+the display to show boot information via the display controller 0 (DC0).
+Nevertheless, when we need to passthrough the DC0 to the guest Display VM,
+first we need to tell the SCU, that this resource is reserved for the VM. When
+we do this, by the host's device tree the U-boot crash on booting. To avoid
+this crash we need to build again the U-boot disabling the video support
+(CONFIG_DM_VIDEO unset).
+
+Signed-off-by: Juan Pablo Ruiz <juanpablo.ruiz@tii.ae>
+---
+ configs/imx8qm_mek_defconfig | 4 ++--
+ include/configs/imx8qm_mek.h | 3 ++-
+ 2 files changed, 4 insertions(+), 3 deletions(-)
+
+diff --git a/configs/imx8qm_mek_defconfig b/configs/imx8qm_mek_defconfig
+index 1ed6eeee57..f272023300 100644
+--- a/configs/imx8qm_mek_defconfig
++++ b/configs/imx8qm_mek_defconfig
+@@ -31,7 +31,7 @@ CONFIG_PANIC_HANG=y
+ CONFIG_OF_SYSTEM_SETUP=y
+ CONFIG_BOOTDELAY=3
+ CONFIG_USE_BOOTCOMMAND=y
+-CONFIG_BOOTCOMMAND="run loadhdp; hdp load ${hdp_addr}; run distro_bootcmd;"
++CONFIG_BOOTCOMMAND="run loadfdt; scu_rm dtb ${fdt_addr}; run loadhdp; hdp load ${hdp_addr}; run distro_bootcmd;"
+ CONFIG_LOG=y
+ CONFIG_BOARD_EARLY_INIT_F=y
+ CONFIG_SPL_BOARD_INIT=y
+@@ -185,7 +185,7 @@ CONFIG_VIDEO_IMX_HDP_LOAD=y
+ CONFIG_OF_LIBFDT_OVERLAY=y
+ 
+ CONFIG_VIDEO_IMXDPUV1=y
+-CONFIG_DM_VIDEO=y
++# CONFIG_DM_VIDEO is not set
+ CONFIG_BMP_16BPP=y
+ CONFIG_BMP_24BPP=y
+ CONFIG_BMP_32BPP=y
+diff --git a/include/configs/imx8qm_mek.h b/include/configs/imx8qm_mek.h
+index ab5b58ec32..28da7675d4 100644
+--- a/include/configs/imx8qm_mek.h
++++ b/include/configs/imx8qm_mek.h
+@@ -197,7 +197,8 @@
+ 	"cntr_addr=0x98000000\0"			\
+ 	"cntr_file=os_cntr_signed.bin\0" \
+ 	"boot_fdt=try\0" \
+-	"fdtfile=imx8qm-mek-hdmi.dtb\0" \
++	"fdtfile=imx8qm-mek-kvm-host.dtb\0" \
++	"fdt_file=imx8qm-mek-kvm-host.dtb\0" \
```

Comment:
Can you please clarify the idea of having two different names for single value?

Reply:
Oh, and also I think you need to change fdt file in `overlays/imx8qm.nix` file

Reply:
Thanks for noting this. I will review and keep only the fdt file that is needed. 

For the overlays/imx8qm.nix you are right, I need to update this with the new fdt file. Let me do this when I integrate the Linux kernel patches that build this fdt file.

Reply:
Sorry, I do not completely understand how this .dtb file is installed to the boot partition. Currently it is done here:
https://github.com/tiiuae/build-configurations/blob/01a40cc360ab7b8d8d6a2a0bae21ecceff53a326/overlays/imx8qm.nix#L46

Reply:
It is not right now, but it will be when I integrate the Linux kernel patches.